### ISSUE TITLE
Titan Ridge support/required?

### ISSUE THREAD

**Issue by tonyarnold** [osy86/ThunderboltReset#2]
Thanks for all of your hard work on getting Thunderbolt working. 

Do you know if similar patches are required (or would work) on Titan Ridge controllers/cards?

**Comment by osy86** [comment]
Preliminary tests shows that it works but I’m out of town for a week so I haven’t really looked into it.

**Comment by tonyarnold** [comment]
Let me know if there's anything you want to try out on a Titan Ridge. I'd bend over backward to have my TB3 display be a little more reliable!

**Comment by noprex** [comment]
I would also love to help out with anything I can with the Titan Ridge, I am far from flashing, considering that I couldn't get the ACPI debug working in order to try the ICM disabling method.

**Comment by tonyarnold** [comment]
Yeah, I also couldn't get ACPI debug logging working. @osy86 is this "Rehabman's ACPI logger"? https://github.com/RehabMan/OS-X-ACPI-Debug

**Comment by northamtrans** [comment]
@tonyarnold this tripped me up at first as well but eventually got it to work. Did you pull your System DSDT, patch it with "add DSDT Debug Methods," then add it to your ACPI along with adding the kext? Also if you get a failed boot after just reset NVRAM from Clover or OC. Helps every time with ACPI changes.

I found my Asus Prime Z370A 2 to be L6F.

If using a version of maciASL that's newer add this resource http://raw.github.com/RehabMan/OS-X-ACPI-Debug/master

**Comment by canyondust** [comment]
@northamtrans The link in the previous reply is broken.

Also, is it necessary to patch the native SSDT? (I would have assumed installing SSDT-HPlog.aml would be enough to load the debug kext as is calls \RMDT.P1)?

EDIT: LINK IS VALID, Browser inserted HTTPS:// (instead of HTTP:)

And I just read the instructions closer. DSDT needs patching as well as debug code where you want it.

**Comment by northamtrans** [comment]
@canyondust that's what I had to do. I pulled mine with Rehabmans Patchmatic then using MACiASL (his version) I added GPE Events. 

![Screen Shot 2020-02-06 at 2 23 52 PM](https://user-images.githubusercontent.com/58002473/73975605-4f693c80-48ec-11ea-89d9-381e15183315.png)
.

**Comment by canyondust** [comment]
Thanks :) I figured my questions out.

**Comment by tonyarnold** [comment]
@canyondust did it work for you? I haven't figured out how to get mine working yet, so I'd love to see the changes you ended up making.

**Comment by canyondust** [comment]
Hasn’t worked for me. All the code works but I get no hardware found by tbpatch, and I haven’t been successful on getting an ACPI event triggered on hot plug with rehabman’s debug. I think this code works fine, but my system might not meet requirements (GA-Z170x-UD5 TH) also the thunderboltreset.kext lists a 10.14 minimum and I’m on 10.13.6. I patched the plist in the kext but no luck there. I also generated individual SSDTs for every GPE candidate for my system and tried them one by one, also no luck :)
